homeassistant/components/litterrobot/select.py

```diff
@@ -42,26 +42,71 @@ class RobotSelectEntityDescription(
     """A class that describes robot select entities."""
 
     entity_category: EntityCategory = EntityCategory.CONFIG
-
-
-LITTER_ROBOT_SELECT = RobotSelectEntityDescription[LitterRobot, int](
-    key="cycle_delay",
-    name="Clean Cycle Wait Time Minutes",
-    icon="mdi:timer-outline",
-    unit_of_measurement=TIME_MINUTES,
-    current_fn=lambda robot: robot.clean_cycle_wait_time_minutes,
-    options_fn=lambda robot: robot.VALID_WAIT_TIMES,
-    select_fn=lambda robot, option: robot.set_wait_time(int(option)),
-)
-FEEDER_ROBOT_SELECT = RobotSelectEntityDescription[FeederRobot, float](
-    key="meal_insert_size",
-    name="Meal insert size",
-    icon="mdi:scale",
-    unit_of_measurement="cups",
-    current_fn=lambda robot: robot.meal_insert_size,
-    options_fn=lambda robot: robot.VALID_MEAL_INSERT_SIZES,
-    select_fn=lambda robot, option: robot.set_meal_insert_size(float(option)),
-)
+    icon_fn: Callable[[_RobotT], str] | None = None
+
+
+ROBOT_SELECT_MAP: dict[type[Robot], tuple[RobotSelectEntityDescription, ...]] = {
+    LitterRobot: (
+        RobotSelectEntityDescription[LitterRobot, int](
+            key="cycle_delay",
+            name="Clean Cycle Wait Time Minutes",
+            icon="mdi:timer-outline",
+            unit_of_measurement=TIME_MINUTES,
+            current_fn=lambda robot: robot.clean_cycle_wait_time_minutes,
+            options_fn=lambda robot: robot.VALID_WAIT_TIMES,
+            select_fn=lambda robot, option: robot.set_wait_time(int(option)),
+        ),
+    ),
+    LitterRobot4: (
+        RobotSelectEntityDescription[LitterRobot4, str](
+            key="night_light_level",
+            name="Night light level",
+            current_fn=lambda robot: None
+            if (level := robot.night_light_level) is None
+            else level.name.capitalize(),
+            options_fn=lambda _: [level.name.capitalize() for level in NightLightLevel],
+            select_fn=lambda robot, option: robot.set_night_light_brightness(
+                NightLightLevel[option.upper()]
+            ),
+            icon_fn=lambda robot: "mdi:lightbulb-question"
+            if (level := robot.night_light_level) is None
+            else "mdi:lightbulb-on"
+            if level == NightLightLevel.HIGH
+            else "mdi:lightbulb-on-70"
+            if level == NightLightLevel.MEDIUM
+            else "mdi:lightbulb-on-30",
```

Comment:
Should this be a separate method? Ternary conditional operator spanning multiple lines may not be preferred.

Reply:
Yeah, stupid brain fart moment. I blame Monday.